engine/src/flutter/impeller/core/texture_descriptor.h

```diff
@@ -80,20 +80,7 @@ struct TextureDescriptor {
     return IsMultisampleCapable(type) ? count > 1 : count == 1;
   }
 
-  constexpr bool operator==(const TextureDescriptor& other) const {
-    return size == other.size &&                          //
-           storage_mode == other.storage_mode &&          //
-           format == other.format &&                      //
-           usage == other.usage &&                        //
-           sample_count == other.sample_count &&          //
-           type == other.type &&                          //
-           compression_type == other.compression_type &&  //
-           mip_count == other.mip_count;
-  }
-
-  constexpr bool operator!=(const TextureDescriptor& other) const {
-    return !(*this == other);
-  }
+  constexpr bool operator==(const TextureDescriptor& other) const = default;
```

Comment:
These default implementations bother me a little. They assume that all fields matter. What if someone adds a new field whose value is not relevant to comparison? This could create a maintenance issue in the future. I would feel happier if the default would take a list of fields to use.

On the other hand, with the old definition there was a potential maintenance issue that someone might add a new field that does matter in comparisons.

6 of one, half a fortnight of the other...

Reply:
Yeah, there were a couple of instances in our codebase where not all were compared. I'm fine assuming everything matters. If it doesn't, we can be explicit about it and mention that in a comment.